Lib/fontTools/ttLib/tables/ttProgram.py

```diff
@@ -248,9 +250,13 @@ def toXML(self, writer, ttFont):
 				writer.begintag("assembly")
 				writer.newline()
 				i = 0
+				indent = 0
 				nInstr = len(assembly)
 				while i < nInstr:
 					instr = assembly[i]
+					if _unindentRE.match(instr):
+						indent -= 1
+					writer.write("  " + "    " * indent)
```

Comment:
@jenskutilek why here are you doing " " + " " * indent and not just " " * indent?
I presume it's because the default indentation for the XMLWriter is two spaces so you are padding it to four...

I think it would be better to stick to whatever XMLWriter.indentwhite value is, i.e. currently two spaces by default, and maybe add a new command-line option for ttx that passes a different indentation setting down to the XMLWriter instance (which already takes a indentwhite=INDENT keyword argument in its constructor).

This way the indentation for the whole xml file would be consistent, i.e. either two spaces everywhere, or four, or tabs, etc. instead of mixing two/four like in here.

WDYT?